### PR TITLE
Add device

### PR DESCRIPTION
**Pull request by nwakauc**
## In this milestone Implemented the following 🧧:

- Remove the `current_user` method in `ApplicationController,` devise will provide us with one.

- Can register a new user.
- The user logs in with a combination of email and password.
- `Hashed` passwords should be stored in the `database.`
- Ask for confirmation by email.
- Can reset password.
- **MY PARTNER @Tekhlay COMMITED ALSO TO THIS REPO**
- **Thank you for reviewing my code 👏.**

### CODE REVIEW

**Comment by addod19** [comment]
### Hi @uchexm, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because this project is a pair programming activity which you did solo. We are expected to see commit messages from both partners

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/user.rb [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-  has_many :comments, foreign_key: 'author_id'
-  has_many :likes, foreign_key: 'author_id'
-
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  has_many :comments, dependent: :destroy, foreign_key: 'author_id'
+  has_many :likes, dependent: :destroy, foreign_key: 'author_id'
+  has_many :posts, dependent: :destroy, foreign_key: 'author_id'
```
Comment (Gambit142) [inline app/models/user.rb:8]: - [ ] When I started your application using the `rails s` command, I got the following error:

image

I noticed that the `devise` gem was not set up properly in this application. I would advise you to read this [article](https://levelup.gitconnected.com/how-to-add-authentication-to-rails-web-app-using-devise-a5caedbbc093) or this [article](https://makingsenseofrails.dev/how-to-set-up-devise-in-rails-7-72626ead7e0b) to set up devise properly in this project. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @uchexm @Tekhlay 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Collaborated with partner ✔️ 
- README.md file is professional ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 class UsersController < ApplicationController
   def index
-    @users = User.all.order(created_at: :asc)
-    @current_user = current_user
+    @users = User.all
   end
```
Comment (Qoosim) [inline app/controllers/users_controller.rb:4]: - [ ] This is what I got while trying to run your app with `rails s`. Kindly check your user controller for the error `undefined method "authenticate_user!"`.  Kindly check this [link](https://github.com/heartcombo/devise) for devise proper setup.
- [ ] You might also need to check your routes.

error

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @uchexm, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**